code/modules/power/cable.dm

```diff
@@ -415,3 +412,172 @@
+		var/disx = spawner.x - src.x
+		var/disy = spawner.y - src.y
+		// the following assumes disxy (displacement of x or y) equals 1,0 or -1
+		if (disx == 1)
+			temp_flags |= EAST
+		if (disx == -1)
+			temp_flags |= WEST
+		if (disy == 1)
+			temp_flags |= NORTH
+		if (disy == -1)
+			temp_flags |= SOUTH
+		// each iteration can only have one direction at a time, luckily
+		if (temp_flags == NORTHEAST)
+			cable_surr |= NE
+		else if (temp_flags == NORTHWEST)
+			cable_surr |= NW
+		else if (temp_flags == SOUTHEAST)
+			cable_surr |= SE
+		else if (temp_flags == SOUTHWEST)
+			cable_surr |= SW
+		else cable_surr |= temp_flags
+	optimise()
+	for (var/obj/cable/normal_cable in orange(1, src))
+	// normal, prexisting, manually placed cables (must be joined to no matter what)
+	// turns out, since initialize() does cablespawners one by one
+	// they turn into regular cables and must be considered like that by the system
+	// this bit is MANDATORY
+		var/temp_flags = 0
+		var/disx = normal_cable.x - src.x
+		var/disy = normal_cable.y - src.y
+		if (disx == 1)
+			temp_flags |= EAST
+		if (disx == -1)
+			temp_flags |= WEST
+		if (disy == 1)
+			temp_flags |= NORTH
+		if (disy == -1)
+			temp_flags |= SOUTH
+		// each iteration can only have one direction at a time, luckily
+		if (temp_flags == NORTHEAST)
+			if (normal_cable.d1 == SOUTHWEST || normal_cable.d2 == SOUTHWEST)
+				cable_surr |= NE
+		else if (temp_flags == NORTHWEST)
+			if (normal_cable.d1 == SOUTHEAST || normal_cable.d2 == SOUTHEAST)
+				cable_surr |= NW
+		else if (temp_flags == SOUTHEAST)
+			if (normal_cable.d1 == NORTHWEST || normal_cable.d2 == NORTHWEST)
+				cable_surr |= SE
+		else if (temp_flags == SOUTHWEST)
+			if (normal_cable.d1 == NORTHEAST || normal_cable.d2 == NORTHEAST)
+				cable_surr |= SW
+		else if (normal_cable.d1 == SOUTH || normal_cable.d2 == SOUTH)
+			cable_surr |= NORTH
+		else if (normal_cable.d1 == NORTH || normal_cable.d2 == NORTH)
+			cable_surr |= SOUTH
+		else if (normal_cable.d1 == WEST || normal_cable.d2 == WEST)
+			cable_surr |= EAST
+		else if (normal_cable.d1 == EAST || normal_cable.d2 == EAST)
+			cable_surr |= WEST
+		// the 'real' wires override and always connect to prevent loose ends
+/// causes cablespawner to spawn cables (amazing)
+/obj/cablespawner/proc/replace()
+	var/directions = convert()
+	if (length(directions) == 0)
+	// a standalone cable (not really supposed to happen if you use cables properly)
+		cable_laying(0, 1)
+	else if (length(directions) == 1)
+	// end of a cable
+		cable_laying(0, directions[1])
+	else if (length(directions) == 2)
+	// a normal, single cable
+		cable_laying(directions[1], directions[2])
+	else if (length(directions) >= 3)
+	// multiple cables, spiral out from the centre
+		for (var/i in 1 to length(directions))
+			cable_laying(0, directions[i])
+	qdel(src)
+/// converts 8 bit into a list of directions. In order of bitflag
+/obj/cablespawner/proc/convert()
```

Comment:
Why not just have cable_surr be the list of directions if you're gonna convert it anyway?

Reply:
i read somewhere that bitflags take up less memory. Plus, this way, in the `optimise()` bit, I can use cool things to do with the builtin diagonal direction flags like `NORTHEAST` and such to do detections.